var/spack/repos/builtin/packages/libxsmm/package.py

```diff
@@ -24,6 +24,7 @@ class Libxsmm(MakefilePackage):
     # 2.0 release is planned for Jan / Feb 2024. This commit from main is added
     # as a stable version that supports other targets than x86. Remove this
     # after 2.0 release.
+    version("main-2024-09", commit="e0c4a2389afba36c453233ad7de07bd92c715bec")
     version("main-2023-11", commit="0d9be905527ba575c14ca5d3b4c9673916c868b2")
     version("main", branch="main")
```

Comment:
```suggestion
    version("main", branch="main")
    version("main-2024-09", commit="e0c4a2389afba36c453233ad7de07bd92c715bec")
    version("main-2023-11", commit="0d9be905527ba575c14ca5d3b4c9673916c868b2")
```

As a floating / infinite version `main` should remain at the top of the versions directives.